inference-engine/src/gna_plugin/gna2_model_dump.cpp

```diff
@@ -0,0 +1,225 @@
+    if (paramCount > 0) {
+        Gna2Shape* subTensorShape = reinterpret_cast<Gna2Shape*>(parameters[CopyShapeParamIdx]);
+        dumpFile << "\tParameter name: shape (sub-tensor shape), ";
+        dumpShape(dumpFile, subTensorShape);
+    }
+}
+
+void dumpFCAffineParameters(std::ostream& dumpFile, void** parameters, int paramCount) {
+    if (paramCount > 0) {
+        Gna2BiasMode* biasMode = reinterpret_cast<Gna2BiasMode*>(parameters[BiasModeFCAffineParamIdx]);
+        if (biasMode != nullptr)
+            dumpFile << "\tParameter name: biasMode, value: " << biasModeMap.find(*biasMode)->second;
+    }
+    if (paramCount > 1) {
+        uint32_t* biasVectorIndex = reinterpret_cast<uint32_t*>(parameters[BiasModeFCAffineParamIdx]);
+        if (biasVectorIndex != nullptr)
+            dumpFile << "\tParameter name: biasVectorIndex, value: " << *biasVectorIndex;
+    }
+}
+
+void dumpGMMParameters(std::ostream& dumpFile, void** parameters, int paramCount) {
+    if (paramCount > 0) {
+        uint32_t* maximumScore = reinterpret_cast<uint32_t*>(parameters[MaximumScoreParamIdx]);
+        dumpFile << "\tParameter name: maximumScore, value: " << *maximumScore;
+    }
+}
+
+void dumpRecurrentParameters(std::ostream& dumpFile, void** parameters, int paramCount) {
+    if (paramCount > 0) {
+        uint32_t* delay = reinterpret_cast<uint32_t*>(parameters[DelayParamIdx]);
+        dumpFile << "\tParameter name: delay, value: " << *delay;
+    }
+}
+
+typedef void (*dumpParamaters) (std::ostream&, void**, int);
+
+const std::map<Gna2OperationType, dumpParamaters> dumpParamMap = {
+    {Gna2OperationTypeConvolution, dumpConvolutionParameters},
+    {Gna2OperationTypeCopy, dumpCopyParameters},
+    {Gna2OperationTypeFullyConnectedAffine, dumpFCAffineParameters},
+    {Gna2OperationTypeGmm, dumpGMMParameters},
+    {Gna2OperationTypeRecurrent, dumpRecurrentParameters}
+};
+
+void dumpPwl(std::ostream & dumpFile, const Gna2Tensor& activation) {
+    const Gna2PwlSegment* const segments = static_cast<Gna2PwlSegment*>(activation.Data);
+    const uint32_t numberOfSegments = activation.Shape.Dimensions[0];
+
+    for (uint32_t k = 0; k < numberOfSegments; k++) {
+        auto scale = ((segments[k].xBase & 3) + 1) * 8;
+        int32_t B = segments[k].xBase & 0xfffffffc;
+
+        dumpFile << "\t\tBase value for input(B) : " << B << "\n";
+        dumpFile << "\t\tBase value for output(b) : " << segments[k].yBase << "\n";
+        dumpFile << "\t\tSegment slope (S): " << segments[k].Slope << "\n";
+        dumpFile << "\t\tShift(scale) : " << scale << "\n";
+        dumpFile << "\t\ty = ax + b:   a = " << segments[k].Slope / (2 ^ scale)
+            << ", b = " << segments[k].yBase - ((B * segments[k].Slope) / (2 ^ scale));
+        if (segments[k].Slope != 0) {
+            dumpFile << ", x0 = " << B - ((segments[k].yBase * (2 ^ scale)) / segments[k].Slope);
+        }
+        dumpFile << "\n";
+    }
+}
```

Comment:
could you provide sample output?

Reply:
I will add a newline after each segment for better readability.

		Base value for input(B) : -2147483648
		Base value for output(b) : 0
		Segment slope (S): 0
		Shift(scale) : 8
		y = ax + b:   a = 0, b = 0
		Base value for input(B) : -21584320
		Base value for output(b) : 0
		Segment slope (S): 8413
		Shift(scale) : 32
		y = ax + b:   a = 247, b = 35301697, x0 = -21584320
		Base value for input(B) : -10707036
		Base value for output(b) : 21
		Segment slope (S): 405
		Shift(scale) : 24
		y = ax + b:   a = 15, b = 1591647, x0 = -10707037